packages/request-manager/src/controller.ts

```diff
@@ -12,23 +12,32 @@ export class TorController extends EventEmitter {
     waitingTime = 1000;
     maxTriesWaiting = 60;
     isCircuitEstablished = false;
+    torIsDisabledWhileStarting: boolean;
 
     constructor(options: TorConnectionOptions) {
         super();
         this.options = options;
+        this.torIsDisabledWhileStarting = false;
         this.controlPort = new TorControlPort(options, this.onMessageReceived.bind(this));
     }
 
     onMessageReceived(message: string) {
         const bootstrap: BootstrapEvent[] = bootstrapParser(message);
-        this.isCircuitEstablished = bootstrap.some(
-            (event: BootstrapEvent) => event.progress === BootstrapEventProgress.Done,
-        );
+        bootstrap.forEach(event => {
+            if (!event || !event.progress) return;
+            this.isCircuitEstablished = event.progress === BootstrapEventProgress.Done;
+            this.emit('bootstrap/event', event);
+        });
     }
 
     waitUntilAlive(): Promise<void> {
         const errorMessages: string[] = [];
+        this.torIsDisabledWhileStarting = false;
         const waitUntilResponse = async (triesCount: number): Promise<void> => {
+            if (this.torIsDisabledWhileStarting) {
+                // If TOR is starting and we want to cancel it.
+                return;
+            }
```

Comment:
previously, this method threw an error in case timeout was reached or resolved in case connection was established

now, it throws error in case timeout is reached and resolves in case connection is established OR torIsDisabledWhileStarting is set to true in the meantime. 

I am just pointing here that this method got resolved but it does not need to mean that tor is alive which is suggested by name of the method. 

so far consider this as a nitpick or maybe a discussion topic

Reply:
Yes, that is right, but we need a way to be able to stop that process while it is happening. Maybe there is other approach, I will think about it.